src/ast.js

```diff
@@ -39,16 +39,22 @@ class CharClass {
     return 1;
   }
 
-  toString() {
-    return this.set.toString();
+  get isSingleCharacter() {
+    return !this.set.toArray().some(c => c > 0xffff);
   }
 
-  getCharClass() {
-    return this.set;
+  get isSingleCodepoint() {
+    return Array.from(this.set).length === 1;
```

Comment:
In unicode mode, character classes can always be treated as a single "codepoint" because that's what they match. In ucs2 mode, regenerate might have to return a sequence instead, which is why we check if any of the values are > 0xffff.